content/en/platform/corda/5.0-beta/developing/development-tutorials/build-basic-cordapp/basic-cordapp-flows.md

```diff
@@ -0,0 +1,796 @@
+---
+date: '2023-01-11'
+title: "Write Flows"
+menu:
+  corda-5-beta:
+    parent: corda-5-beta-tutorial-develop-building-your-first-basic-cordapp
+    identifier: corda-5-beta-tutorial-develop-write-flows
+    weight: 1400
+section_menu: corda-5-beta
+---
+
+In Corda, flows automate the process of agreeing ledger updates. They are a sequence of steps that tell the node how to achieve a specific ledger update, such as issuing an asset or making a deposit. Nodes communicate using these flows in point-to-point interactions, rather than a global broadcast system. Network participants must specify what information needs to be sent, to which counterparties.
+
+This tutorial guides you through writing the three flows you need in your CorDapp. These are:
+
+* `CreateAndIssueAppleStampFlow`
+* `PackageApplesFlow`
+* `RedeemApplesFlow`
+
+You will be creating these flows in the `ledger-utxo-example-apples-app/src/main/kotlin/net/cordapp/utxo/apples/flows` directory in this tutorial.
+
+## Learning Objectives
+
+After you have completed this tutorial, you will know how to write and implement flows in a CorDapp.
+
+## Before You Start
+
+Before you start writing flows, read more about [flows](../../ledger/flows.md).
+
+## Write the `CreateAndIssueAppleStampFlow`
+
+The `CreateAndIssueAppleStampFlow` creates the `AppleStamp` and issues it to the customer.
+
+### Write the Initiator Flow
+
+The `CreateAndIssueAppleStampFlow` action requires interaction between the issuer and the customer. For this reason, you must create an initiator flow and a responder flow.
+
+
+#### Implement the `CreateAndIssueAppleStampFlow` class
+
+Add the `CreateAndIssueAppleStampFlow` class to `net/cordapp/utxo/apples/flowsnet/cordapp/utxo/apples/flows/issue`.
+
+
+#### Make the `CreateAndIssueAppleStampFlow` Startable by RPC
+
+Add the `RPCStartableFlow` to the `CreateAndIssueAppleStampFlow`.
+
+This allows the flow to be started by RPC. You must implement this interface if you want to trigger the flow with RPC via Corda’s HTTP endpoints.
+
+So far your code should look like this:
+
+```kotlin
+class CreateAndIssueAppleStampFlow : RPCStartableFlow
+```
+
+#### Allow `CreateAndIssueAppleStampFlow` to Initiate Flows with Peers
+
+Add the `@InitiatingFlow` annotation to `CreateAndIssueAppleStampFlow`.
+
+This indicates that this flow is the initiating flow in an initiating and initiated flow pair. A `protocol` must be defined that both the initiating and initiated flow reference.
```

Comment:
@lankydan Sounds like an incomplete sentence. Is it "...defined that both" or "...defined in both"?

Reply:
I think either work, trying to show that you need flows that look like:

```kotlin
@InitiatingFlow(protocol = "something")
class MyFlow ....

@InitiatedBy(protocol = "something")
class MyResponderFlow ...
```

Where the "something" must be the same in both.

Reply:
Rephrased.